src/main/java/org/kohsuke/github/GitHubRequest.java

```diff
@@ -0,0 +1,651 @@
+         * @return the request builder
+         */
+        public B with(String key, long value) {
+            return with(key, (Object) value);
+        }
+
+        /**
+         * With requester.
+         *
+         * @param key
+         *            the key
+         * @param value
+         *            the value
+         * @return the request builder
+         */
+        public B with(String key, boolean value) {
+            return with(key, (Object) value);
+        }
+
+        /**
+         * With requester.
+         *
+         * @param key
+         *            the key
+         * @param e
+         *            the e
+         * @return the request builder
+         */
+        public B with(String key, Enum<?> e) {
+            if (e == null)
+                return with(key, (Object) null);
+            return with(key, transformEnum(e));
+        }
+
+        /**
+         * With requester.
+         *
+         * @param key
+         *            the key
+         * @param value
+         *            the value
+         * @return the request builder
+         */
+        public B with(String key, String value) {
+            return with(key, (Object) value);
+        }
+
+        /**
+         * With requester.
+         *
+         * @param key
+         *            the key
+         * @param value
+         *            the value
+         * @return the request builder
+         */
+        public B with(String key, Collection<?> value) {
+            return with(key, (Object) value);
+        }
+
+        /**
+         * With requester.
+         *
+         * @param key
+         *            the key
+         * @param value
+         *            the value
+         * @return the request builder
+         */
+        public B with(String key, Map<?, ?> value) {
+            return with(key, (Object) value);
+        }
+
+        /**
+         * With requester.
+         *
+         * @param body
+         *            the body
+         * @return the request builder
+         */
+        public B with(@WillClose /* later */ InputStream body) {
+            this.body = body;
+            return (B) this;
+        }
+
+        /**
+         * With nullable requester.
+         *
+         * @param key
+         *            the key
+         * @param value
+         *            the value
+         * @return the request builder
+         */
+        public B withNullable(String key, Object value) {
+            args.add(new Entry(key, value));
+            return (B) this;
+        }
+
+        /**
+         * With requester.
+         *
+         * @param key
+         *            the key
+         * @param value
+         *            the value
+         * @return the request builder
+         */
+        public B with(String key, Object value) {
+            if (value != null) {
+                args.add(new Entry(key, value));
+            }
+            return (B) this;
+        }
+
```

Comment:
This is a somewhat arbitrary place for me to put this question, but it's as good as any. There are a lot of `with(thing, thing)` methods here with different parameters. Why is that? I mean, I'm sure there's a perfectly good reason for it, I just am having trouble putting it together on my own.

Reply:
I'm not sure.  These were here when I arrived.  I'm not sure it they are still all needed, per se.  I think I tried to remove them at one point and started getting errors ... but I don't remember now.

Reply:
Opened #699 to come back to this.